src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -1049,10 +1111,13 @@ define(function(require) {
 
 		/**
 		 * Get the template for Ownership Confirmation step (bill upload view)
-		 * @param  {Object} args  Wizard args
+		 * @param  {Object} args
+		 * @param  {Object} args.data  Wizard data
+		 * @param  {Object} args.invoiceDocumentMetadata Invoice document metadata
 		 */
 		portWizardOwnershipConfirmationBillUploadGetTemplate: function(args) {
 			var self = this,
+				invoiceDocumentMetadata = args.invoiceDocumentMetadata,
```

Comment:
Not sure why `invoiceDocumentMetadata` gets passed down two methods when `portWizardOwnershipConfirmationBillUploadBindEvents` is the only one that makes use of it.

Reply:
It is used at `portWizardOwnershipConfirmationBillUploadBindEvents` and `portWizardOwnershipConfirmationAccountInfoBindEvents`. As the information is obtained from the submodule store, do you think it would be better to get the invoice document metadata whithin each method, instead of passing it all the way down from `portWizardOwnershipConfirmationRender`?

Reply:
> [...] do you think it would be better to get the invoice document metadata whithin each method, instead of passing it all the way down from `portWizardOwnershipConfirmationRender`?

From what I can tell, passing down `invoiceDocumentMetadata` adds complexity that could be avoided since that metadata is simply a map of static data.

I believe the problem stems from how `requiredDocuments.documents` is structured. Switching from an array to a map would make more sense and make it easier to retrieve the attachment name (you could use `lodash#get` or make a util for it to abstract the path).

If the order of `requiredDocuments.documents` matters, then it should be broken down into two different data structures: one array to hold the order of documents IDs and another to map document IDs to attachment name.